third_party/chromium/media/formats/mp4/box_reader.h

```diff
@@ -81,6 +81,31 @@ class MEDIA_EXPORT BufferReader {
   // Advance the stream by this many bytes.
   bool SkipBytes(uint64_t nbytes) WARN_UNUSED_RESULT;
 
+#if defined(STARBOARD)
+  // Algorithm from the AV1 specification.
+  // https://aomediacodec.github.io/av1-spec/#leb128.
+  template <typename T>
+  bool ReadLeb128(uint8_t* buffer, T* encoded_value, int* bytes_read,
+                  const int max_encoded_bytes = sizeof(uint64_t)) {
+    CHECK(buffer);
+    CHECK(encoded_value);
+    CHECK(bytes_read);
+    RCHECK(sizeof(encoded_value) >= max_encoded_bytes);
+    *encoded_value = 0;
+    *bytes_read = 0;
+    for (int i = 0; i < max_encoded_bytes; ++i) {
+      uint8_t* byte = &buffer[i];
+      RCHECK(Read1(byte));
+      *encoded_value |= ((*byte & 0x7f) << (i * 7));
```

Comment:
I see what you're saying. Is it possible the rvalue returned by `((*byte & 0x7f) << (i * 7))` is only a byte wide? I assumed it would resolve to an int large enough to hold the value.